ui/lib/kv/addon/components/page/list.js

```diff
@@ -43,6 +43,15 @@ export default class KvListPageComponent extends Component {
     return pathIsDirectory(this.secretPath) ? 'View directory' : 'View secret';
   }
 
+  // callback from HDS pagination to set the queryParams currentPage
+  get paginationQueryParams() {
+    return (page) => {
+      return {
+        currentPage: page,
```

Comment:
We probably also want to reset currentPage here if pageSize changes, right? and what happens if there's a filter applied but you change pages?

Reply:
We should probably test/fix this before merging

Reply:
Ah, I missed that. Good catch! It errors without a console error or anything. Just becomes unclickable. Let me fix.

Reply:
Okay, that was actually a bug from the merge. Now back in working state. I tested changing the currentPageSize and I'm pretty sure we don't have to keep track of it because it's always set by lazyPaginatedQuery -[ see line here](https://github.com/hashicorp/vault/blob/main/ui/app/services/store.js#L75). I've tested by overriding this value with say 19 and it works as expected. The only value the user can change via the ui is the currentPage.

Here is a gif of me using pageFilter and clicking through pages. Seems to work as I would expect. Thoughts?

https://github.com/hashicorp/vault/assets/6618863/ba6a96ca-9778-4dee-ac34-166d364e35ce

Reply:
Oh right, if you don't explicitly reset a query param, ember will keep it when you call a transition with other query params. Great, thanks for checking!